pkg/directory/types/farm.go

```diff
@@ -119,6 +119,11 @@ func (f FarmFilter) WithFarmQuery(q FarmQuery) FarmFilter {
 
 }
 
+// WithIP filter farm ips by specific ip
+func (f FarmFilter) WithIP(ip generated.FarmerIP) FarmFilter {
+	return append(f, bson.E{Key: "ip", Value: ip})
```

Comment:
This is wrong because now you also searching WITH the reservation id (as it is part of FarmerIP) so unless you have exact value this match will not work

Please check how you can search inside the element of the array. as far as i remember it will be something like `Key: "ip.ip"`

Q: is this filter even needed ?

Reply:
This filter makes sure you get the IP with the expected value for it. This will counter updating the IP when another client already updated it.

In example, when I make a reservation for a public IP we expect the reservation ID to be 0 because this IP was not reserved yet. If we find it we update it. If another client comes in the reserve the same ip this query will not find this ip with reservation 0 because we already updated it before!